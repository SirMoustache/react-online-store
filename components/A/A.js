/**
 * Absolute imports
 */
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * Utils
 */
import { withDynamicComponent } from '../../utils/styledComponentUtils';

/**
 * Theme
 */
import theme from '../../theme';

/**
 * A link to a certain page, an anchor tag
 */
const A = styled.a`
  color: ${theme.colors.active.main};
  cursor: pointer;
  text-decoration: none;

  &:hover {
    color: ${theme.colors.active.dark};
  }
`;

A.propTypes = {
  /**
   * The component used for the root node.
   * Either a string to use a DOM element or a component.
   */
  component: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
    PropTypes.object,
  ]),
  href: PropTypes.string,
  title: PropTypes.string,
  rel: PropTypes.string,
  target: PropTypes.string,
};

export default withDynamicComponent(A);
