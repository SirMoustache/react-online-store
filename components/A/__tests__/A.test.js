/**
 * Absolute imports
 */
import React from 'react';
import { shallow } from 'enzyme';
// import { enzymeFind } from 'styled-components/test-utils';

/**
 * Local Components
 */
import A from '..';

describe('<A />', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<A />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render children when passed in', () => {
    const child = <span className="unique" />;
    const wrapper = shallow(
      <A>
        <child />
      </A>,
    );
    expect(wrapper.contains(<child />)).toEqual(true);
  });

  it('should adopt a valid attribute', () => {
    const href = '/';
    const wrapper = shallow(<A href={href} />);
    expect(wrapper.prop('href')).toEqual(href);
  });

  it('should render as provided component', () => {
    const component = 'button';
    const wrapper = shallow(<A component={component} />);
    expect(wrapper.find(component)).toBeTruthy();
  });
});
