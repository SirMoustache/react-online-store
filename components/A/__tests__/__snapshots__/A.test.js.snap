// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<A /> should match snapshot 1`] = `<A />`;
